packages/typegen/src/write/inline.ts

```diff
@@ -69,31 +73,51 @@ export function getFileWriter({getQueriesModulePath = defaultGetQueriesModule, w
 
     await writeFile(file, newSource)
 
-    function visit(node: ts.Node) {
-      if (ts.isModuleDeclaration(node) && node.name.getText() === 'queries') {
+    function visitRecursive(node: ts.Node) {
+      if (ts.isModuleDeclaration(node) && node.name.getText() === queryNamespace) {
+        // remove old import(s) (will get re-added later)
         edits.push({
           start: node.getStart(sourceFile),
           end: node.getEnd(),
           replacement: '',
         })
+        return
       }
 
       if (ts.isTaggedTemplateExpression(node)) {
-        const isSqlIdentifier = (n: ts.Node) => ts.isIdentifier(n) && n.getText() === 'sql'
-        const sqlPropertyAccessor = ts.isPropertyAccessExpression(node.tag) && isSqlIdentifier(node.tag.name)
-        if (isSqlIdentifier(node.tag) || sqlPropertyAccessor) {
-          const match = group.find(q => q.text === node.getFullText())
-          if (match) {
+        const isSqlIdentifier = (e: ts.Node) => ts.isIdentifier(e) && e.getText() === 'sql'
+        const isSqlPropertyAccessor = (e: ts.Expression) => ts.isPropertyAccessExpression(e) && isSqlIdentifier(e.name)
+        if (!isSqlIdentifier(node.tag) && !isSqlPropertyAccessor(node.tag)) {
+          return
+        }
+        const matchingQuery = group.find(q => q.text === node.getFullText())
+        if (!matchingQuery) {
+          return
+        }
+        const typeReference = `${queryNamespace}.${matchingQuery.tag}`
+        if (node.typeArguments && node.typeArguments.length === 1) {
+          // existing type definitions
+          const [typeNode] = node.typeArguments
+          if (ts.isIntersectionTypeNode(typeNode)) {
+            // we want to preserve intersection types
+            const [firstArg] = typeNode.types // We can't be sure the first argument is a generated type, but as the namespace might have been overwritten we're gonna have to assume.
```

Comment:
Minor: I don't think this comment reflects (what I think of) as slonik-typegen's general attitude. Even if it's not a generated type, when I'm a user I _want_ slonik typegen to take control and overwrite. This intersection trick and the ignore feature coming up are escape hatches because slonik-typegen, slonik, and postgres itself are imperfect. So I'd suggested removing or making the comment more "aggressive" to show that we're doing it on purpose in case a future person thinks it's a mistake:
```suggestion
            // Always overwrite the first type in the intersection, leave the second alone.
            const [firstArg] = typeNode.types
```

If we make the `queries` namespace configurable in future we could consider making this smarter and less aggressive but until then this feels more honest to me.

Reply:
I Agree with your point but not with your change.

The first comment is meant to signal to the reader what happens in a specific scenario/routine.
I often do this at the beginning of `if` clauses to signal which case is supposed to be handled here.
Compare: https://github.com/mmkal/slonik-tools/pull/389/files#diff-971380bf36250a35a251a205f99aed60a80f9b5153deacde49eca03408d87975R78

The second comment should be made more "aggressive", as you put it, but it doesn't only leave the second alone, but all subsequent intersections.

Reply:
```suggestion
            // preserve intersection types
            const [firstArg] = typeNode.types // Always overwrite the first type in the intersection, leave all subsequent second alone.
```

Reply:
I’m good w that!